.github/workflows/ci.yaml

```diff
@@ -28,51 +32,44 @@ jobs:
           - stm32mp1
           - stm32wl
           - stm32wb
+
     env:
       CRATES: ${{ matrix.crate }}
       CARGO_INCREMENTAL: 0
+      SVDTOOLS: svdtools
+
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
+
       - name: Install Rust
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
           profile: minimal
           override: true
           components: rustfmt
-      - name: Install xmllint
-        run: sudo apt-get install libxml2-utils
-      - name: Install Python dependencies
-        run: |
-          pip3 install --user setuptools wheel
-          pip3 install --user svdtools
-      - name: Put pip binary directory into path
-        run: echo "~/.local/bin" >> $GITHUB_PATH
-      - name: Cache Cargo installed binaries
-        uses: actions/cache@v1
-        id: cache-cargo
-        with:
-          path: ~/cargo-bin
-          key: ${{ runner.os }}-svd2rust-0.21.0
-      - name: Install svd2rust
-        if: steps.cache-cargo.outputs.cache-hit != 'true'
-        uses: actions-rs/install@v0.1
+
+      - name: Create working crate
+        run: make crates
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
         with:
-          crate: svd2rust
-          version: 0.21.0
-      - name: Copy svd2rust to cache directory
-        if: steps.cache-cargo.outputs.cache-hit != 'true'
+          working-directory: ${{ matrix.crate }}
+          key: svdtools-${{ env.SVDTOOLS_VERSION }}-svd2tust-${{ env.SVD2RUST_VERSION }}
+
+      - name: Install tools
         run: |
-          mkdir ~/cargo-bin
-          cp ~/.cargo/bin/svd2rust ~/cargo-bin
-      - name: Put new cargo binary directory into path
-        run: echo "~/cargo-bin" >> $GITHUB_PATH
+          cargo install svdtools --version $SVDTOOLS_VERSION --target-dir deps-target
+          cargo install svd2rust --version $SVD2RUST_VERSION --target-dir deps-target
+          rm -rf deps-target
+
       - name: Patch SVDs
         run: make -j2 patch
-      - name: Check SVDs
-        run: make lint
```

Comment:
I'd rather keep the lint step if we can, why remove it?

Reply:
Why? It's only wastes time and do nothing already made by `svd_encode::Encode` (formats during encoding) and `svd2rust --strict` (makes a lot of checks)

Reply:
It checks the SVDs definitely comply with the published SVD specification though, which has caught errors in tooling before and might again in the future. If it passes we know the patched SVDs are valid, even if they're not going to be used by svd2rust.

Reply:
bors just checks CI (we don't have a separate CI job just for bors now) and there's no separate CI on publish either, so having it here just makes sure it does pass with bors or publish. Is it very slow?